### DISCUSSION TITLE
How to set dark mode by default

### DISCUSSION THREAD

**Discussion by nguyenyou** [timlrx/tailwind-nextjs-starter-blog#222] [Q&A] [answered]
Could you help me to set the default theme is dark?

I've tried to set `defaultTheme` to `dark` for ThemeProvider, but the switch theme button appears to be inconsistent:

```jsx
<ThemeProvider attribute="class" defaultTheme="dark">
```

Would you please help me to do this the right way?

**Comment by timlrx** [comment, marked as answer]
I think you need to set `enableSystem` to `false` too. Do note that once you have modified the themeSwitch, it will save the preferred theme to local storage and this will take priority over the `defaultTheme` you specify. So if you are trying things out, you might want to change the `storageKey`.
